tests/generator/CodeBlockTests.cs

```diff
@@ -0,0 +1,195 @@
+using System.Collections.Generic;
+using System.IO;
+using NUnit.Framework;
+
+namespace GeneratorTests {
+	public class CodeBlockTests {
+		string PerformWriting (ICodeBlock codeBlock)
+		{
+			using var writer = new StringWriter ();
+			codeBlock.Print (writer);
+			return writer.ToString ();
+		}
+
+		[Test]
+		public void LineBlockTest ()
+		{
+			string expectedText =
+				"{\n" +
+				"    using System;\n" +
+				"    using Network;\n" +
+				"}\n";
+			CodeBlock codeBlock = new (new string [] { "using System;", "using Network;" });
+			string output = PerformWriting (codeBlock);
+			Assert.AreEqual (expectedText, output);
+
+			expectedText =
+				"{\n" +
+				"    using System;\n" +
+				"    using Network;\n" +
+				"    using NUnit;\n" +
+				"}\n";
+			codeBlock.AddLine ("using NUnit;");
+			output = PerformWriting (codeBlock);
+			Assert.AreEqual (expectedText, output);
+		}
+
+		[Test]
+		public void MethodBlockAddLinesTest ()
+		{
+			string inputText1 = "int fooCount = 1;";
+			string inputText2 = "string[] fooNames = new [] {\"foo\"};";
+			string methodName = "public void Foobinate";
+			string [] methodArguments = new [] { "int count", "bool isFoo" };
+			string expectedText =
+				"public void Foobinate(int count, bool isFoo)\n" +
+				"{\n" +
+				"    int fooCount = 1;\n" +
+				"    string[] fooNames = new [] {\"foo\"};\n" +
+				"}\n";
+			MethodBlock methodBlock = new (methodName, methodArguments);
+			methodBlock.AddLine (inputText1);
+			methodBlock.AddLine (inputText2);
+			string output = PerformWriting (methodBlock);
+			Assert.AreEqual (expectedText, output);
+		}
+
+		[Test]
+		public void MethodBlockTest ()
+		{
+			string expectedText =
+				"public void Foobinate(int count, bool isFoo)\n" +
+				"{\n" +
+				"    int fooCount = 1;\n" +
+				"    string[] fooNames = new [] {\"foo\"};\n" +
+				"}\n";
+			List<ICodeBlock> blocks = new List<ICodeBlock> () { new LineBlock ("int fooCount = 1;"), new LineBlock ("string[] fooNames = new [] {\"foo\"};") };
+			MethodBlock methodBlock = new ("public void Foobinate", blocks, "int count", "bool isFoo");
```

Comment:
Like above, something like this would be easier to read (one reason being that the layout here is in the same order the code will be generated, you don't have to reorder chunks of text in your head to make sense of it):

```cs
MethodBlock methodBlock = new ("public void Foobinate", "int count", "bool isFoo") {
	new LineBlock ("int fooCount = 1;"),
	new LineBlock ("string [] ...;"),
};
```

Just to be clear: this is how I'd like to use the new API in the generator, and the idea is to change the API to make something like this possible.